src/datatypes/dataset.ts

```diff
@@ -1025,16 +1026,28 @@ export class Dataset implements Instance<DatasetValue, DatasetJS> {
     return new Dataset(value);
   }
 
+  public sameKeys(other: Dataset): boolean {
+    return this.keys.join('|') === other.keys.join('|');
+  }
+
+  public getKeyValueForDatum(datum: Datum): string {
+    const { keys } = this;
+    if (!keys) throw new Error('join lhs must have a key (be a product of a split)');
+    return this.keys.map(k => {
+      let v: any = datum[k];
+      if (v && v.start) v = v.start;
+      if (v && v.toISOString) v = v.toISOString();
+      return v;
+    }).join('|');
```

Comment:
Is there any particular significance to using pipe as the delimiter or could it just as well be anything?

Reply:
nope, no significance